schema/encoding/jsonschema/jsonschema.go

```diff
@@ -41,7 +44,25 @@ func (ew errWriter) write(b []byte) {
 	_, ew.err = ew.w.Write(b)
 }
 
-// ValidatorToJSONSchema takes a validator and renders to JSON
+// boundariesToJSONSchema writes JSON Schema keys and values based on b, prefixed by a comma and without curly braces,
+// to w. The prefixing comma is only written if at least one key/value pair is also written.
+func boundariesToJSONSchema(w io.Writer, b *schema.Boundaries) error {
+	if b == nil {
+		return nil
+	}
+	ew := errWriter{w: w}
+
+	if !math.IsNaN(b.Min) && !math.IsInf(b.Min, -1) {
+		ew.writeFormat(`, "minimum": %s`, strconv.FormatFloat(b.Min, 'E', -1, 64))
+	}
+	if !math.IsNaN(b.Max) && !math.IsInf(b.Max, 1) {
+		ew.writeFormat(`, "maximum": %s`, strconv.FormatFloat(b.Max, 'E', -1, 64))
+	}
+	return ew.err
+}
```

Comment:
I added this method to use the same code-path for encoding both integer and float boundaries. I also added support for NaN and Inf values so that:

- Max=NaN is treated as no limit.
- Max=+Inf is treated as no limit.
- Min=NaN is treated as no limit.
- Min=-Inf is treated as no limit.

Might be useful in some cases to only set one limit... I checked that this worked with the existing schema package logic by first inspecting the code, and then by adding some tests.

Hint: The following is always true for NaN, where f is any float64 value:
- `(f < NaN) == false`
- `(f > NaN) == false`